### ISSUE TITLE
Tests: Pre-populated FakeProductRepository?

### ISSUE THREAD

**Issue by jorbor** [olgakl/OlgasInventorySystem#1]
Tests assume a pre-populated product repository. What do you see as benefits/drawbacks of removing pre-populated data and moving setup of repository to be more local to tests?

**Comment by olgakl** [comment]
Here I tried to simulate having some data already there (if you started off
trying to do a GET, I wanted you to have some data already present).
However, I could have made this optional by having one constructor without
pre populated data and another constructor that took in data to pre
populate with as a parameter.

**Comment by jorbor** [comment]
Thanks for the response, Olga!

I'll be a little more specific: having data set up in a separate file means we have two places to look when writing a test (repository + test) , rather than just one (test).

In your view, is one of these preferable? Are you familiar with the law of demeter?

**Comment by olgakl** [comment]
Yes, I am familiar with the law of demeter. I pre-populated data in the
constructor before as an example. However, I wouldn't have done that in
production code. Since you saw that as a concern, I updated my repository
with some changes, please take a look.
